sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ParallelTransferOptions.java

```diff
@@ -61,17 +65,22 @@ public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressRe
      * service accepts for uploading in a single requests and is represented by
      * {@link BlockBlobAsyncClient#MAX_UPLOAD_BLOB_BYTES}.
      */
-    public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressReceiver progressReceiver,
-        Integer maxSingleUploadSize) {
+    public ParallelTransferOptions(Integer blockSize, Integer maxConcurrency, ProgressReceiver progressReceiver,
+            Integer maxSingleUploadSize) {
         if (blockSize != null) {
             StorageImplUtils.assertInBounds("blockSize", blockSize, 1, BlockBlobAsyncClient.MAX_STAGE_BLOCK_BYTES);
         }
         this.blockSize = blockSize;
 
-        if (numBuffers != null) {
-            StorageImplUtils.assertInBounds("numBuffers", numBuffers, 2, Integer.MAX_VALUE);
+        /*
+        In buffered upload cases, maxConcurrency must be at least 2 as it also indicates the number of buffers we will
+        allocate. That minimum is validated in the UploadBufferPool. Because this value is also used in file transfers
+        which do not buffer, we only check the absolute minimum for this value here.
```

Comment:
Or we can solve this problem by allocating `maxConcurrency+1` buffers in the UploadbufferPool.